pkg/cloud/vsphere/services/govmomi/update.go

```diff
@@ -42,8 +42,14 @@ func Update(ctx *context.MachineContext) error {
 	if err != nil {
 		return err
 	}
+
+	// If no VM was found then it means the machine has a MachineRef that refers
+	// to a VM that no longer exists. Clear the existing MachineRef and requeue
+	// this machine so that a new VM is created.
 	if vm == nil {
-		return errors.Errorf("vm should already exist for %q", ctx)
+		ctx.MachineConfig.MachineRef = ""
+		ctx.Logger.V(2).Info("vm does not exist; requeue for creation")
```

Comment:
Maybe make this error message more specific to the condition, like “VM was deleted by an external entity, recreating”

Reply:
Hi @frapposelli,

I'm more than happy to make the error more descriptive, but truthfully we don't know *why* or *how* or by *whom* the VM was removed. We just know it's no longer there. I'm all for adding some code later to try and find a task/event that provides additional information related to the deletion event.

Reply:
Hi @frapposelli,

I think I *should* rename this to something like "vm cannot be found; requeue for creation". Because truthfully the VM *could* exist, but we don't have access to it or its ID has changed. I may even want to use this PR to update the permanent ID to the BIOS UUID instead of the MoRef CAPV currently uses.

Reply:
If I'm looking at the CAPV log, I would be fine with just knowing that the condition that triggered a recreate is that the vm *existed, but is no longer there*. More details are absolutely welcome though.

Reply:
Hi @frapposelli,

Fair. I think I'll just update the log for now then, since anything more requires some thinking and quite a bit of work for what you originally requested.

Reply:
Hi @frapposelli,

Please check now, thanks!

Reply:
👍